docs/changelog.md

```diff
@@ -5,6 +5,343 @@ description: |
+  - Fixed baseline of large roots
+  - Fixed multiple minor layout bugs with attachments
+  - Fixed that alignment points could affect line height in math
+  - Fixed that spaces could show up between text and invisible elements like
+    [`metadata`] in math
+  - Fixed a crash with recursive show rules in math
+
+- Introspection
+  - Implemented a new system by which Typst tracks where elements end up on the
+    pages. This may lead to subtly different behaviour in introspections.
+    (**Breaking change**)
+  - Fixed various bugs with wrong counter behaviour in complex layout
+    situations, through a new, more principled implementation
+  - Counter updates can now be before the first, in between, and after the last
+    page when isolated by weak page breaks. This allows, for instance, updating
+    a counter before the first page header and background.
+  - Fixed incorrect [`here().position()`]($here) when [`place`] was used in a
+    context expression
+  - Fixed resolved positions of elements (in particular, headings) whose show
+    rule emits an invisible element (like a state update) before a page break
+  - Fixed behaviour of stepping a counter at a deeper level that its current
+    state has
+  - Fixed citation formatting not working in table headers and a few other
+    places
+  - Displaying the footnote counter will now respect the footnote numbering
+    style
+
+- Model
+  - Document set rules do not need to be at the very start of the document
+    anymore. The only restriction is that they must not occur inside of layout
+    containers.
+  - The `spacing` property of [lists]($list.spacing),
+    [enumerations]($enum.spacing), and [term lists]($terms.spacing) is now also
+    respected for tight lists
+  - Tight lists now only attach (with tighter spacing) to preceding paragraphs,
+    not arbitrary blocks
+  - The [`quote`] element is now locatable (can be used in queries)
+  - The bibliography heading now uses `depth` instead of `level` so that its
+    level can still be configured via a show-set rule
+  - Added support more [numbering] formats: Devanagari, and Eastern Arabic, and
+    Circled Number
+  - Added [`hanging-indent`]($heading.hanging-indent) parameter to heading
+    function to tweak the appearance of multi-line headings and improved default
+    appearance of multi-line headings
+  - Improved handling of bidirectional text in outline entry
+  - Fixed that document set rules were ignored in an otherwise empty document
+  - Fixed that document set rules could not be used in context expressions
+  - Fixed bad interaction between `{set document}` and `{set page}`
+  - Fixed `{show figure: set align(..)}`. Since the default figure alignment is
+    now a show-set rule, it is not revoked by `{show figure: it => it.body}`
+    anymore. (**Minor breaking change**)
+  - Fixed numbering of footnote references
+
+- Visualization
+  - Added non-zero and even-odd fill rules to [`path`] and [`polygon`]
```

Comment:
Use an en-dash for “even–odd”:

```suggestion
  - Added non-zero and even–odd fill rules to [`path`] and [`polygon`]
```

Reply:
Debatable. It's best if this reflects the exact parameter name which will only use a hyphen and it's more of a compound word than, what I assume you based this decision on, something that can be connected with "to" --- I can see why it could work though.

Reply:
Pretty sure this wouldn't match the actual docs for the parameter. I think it's fine as is (with the added benefit of being copy-pastable). But that's just my opinion, in practice it doesn't matter too much as not many people will get this far in the changelog, as I'd say the highlights section is already enough for most :sweat_smile:

Reply:
The other suggestions are good, but with this one I would also rather leave it as-is.